src/event_handler.rs

```diff
@@ -258,59 +220,40 @@ pub fn run_program_on_player_events(event: PlayerEvent, onevent: &str) {
                 env_vars.insert("PLAYER_EVENT", "volume_set".to_string());
                 env_vars.insert("VOLUME", volume.to_string());
             }
-            PlayerEvent::Seeked { new_position_ms } => {
+            PlayerEvent::Seeked { position_ms } => {
                 env_vars.insert("PLAYER_EVENT", "seeked".to_string());
-                env_vars.insert("POSITION_MS", new_position_ms.to_string());
+                env_vars.insert("POSITION_MS", position_ms.to_string());
             }
-        }
-
-        let mut v: Vec<&str> = onevent.split_whitespace().collect();
-
-        info!("Running {:?} with environment variables {:?}", v, env_vars);
-
-        AsyncCommand::new(&v.remove(0))
-            .args(&v)
-            .envs(env_vars.iter())
-            .spawn()
-    };
-
-    emit_non_blocking_event(result());
-}
-
-pub fn run_program_on_session_events(event: SessionEvent, onevent: &str) {
-    let result = || {
-        let mut env_vars = HashMap::new();
-        match event {
-            SessionEvent::Connected {
+            PlayerEvent::SessionConnected {
                 connection_id,
                 user_name,
             } => {
-                env_vars.insert("SESSION_EVENT", "connected".to_string());
+                env_vars.insert("PLAYER_EVENT", "session_connected".to_string());
                 env_vars.insert("CONNECTION_ID", connection_id);
                 env_vars.insert("USER_NAME", user_name);
             }
-            SessionEvent::Disconnected {
+            PlayerEvent::SessionDisconnected {
                 connection_id,
                 user_name,
             } => {
-                env_vars.insert("SESSION_EVENT", "disconnected".to_string());
+                env_vars.insert("SESSION_EVENT", "session_disconnected".to_string());
```

Comment:
```suggestion
                env_vars.insert("PLAYER_EVENT", "session_disconnected".to_string());
```

You probably missed this one. :)